src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item) {
+            this.item = item;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> temporaryItem = tail;
+        Node<T> newNode = new Node<>(value);
+        tail = newNode;
+        if (temporaryItem == null) {
+            head = newNode;
+        } else {
+            temporaryItem.next = newNode;
+            newNode.prev = temporaryItem;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> temporaryItem = head;
+            Node<T> newNode = new Node<>(value);
+            for (int i = 0; i <= index; i++) {
+                if (i == index) {
+                    if (temporaryItem == head) {
+                        temporaryItem.prev = newNode;
+                        newNode.next = temporaryItem;
+                        head = newNode;
+                    } else {
+                        temporaryItem.prev.next = newNode;
+                        newNode.prev = temporaryItem.prev;
+                        temporaryItem.prev = newNode;
+                        newNode.next = temporaryItem;
+                    }
+                    size++;
+                }
+                temporaryItem = temporaryItem.next;
```

Comment:
To improve performance, you don't need to iterate from the head to the given index every time. You can iterate from the head or tail depending on which one is closer to the index.